src/Compilers/CSharp/Portable/FlowAnalysis/FlowAnalysisPass.cs

```diff
@@ -41,9 +47,27 @@ internal class FlowAnalysisPass
                 // will be reported by the lambda binder.
                 Debug.Assert(method.MethodKind != MethodKind.AnonymousFunction);
 
+                // Add implicit "return default(T)" if this is a submission that does not have a trailing expression.
+                var submissionResultType = (method as SynthesizedInteractiveInitializerMethod)?.ResultType;
+                if (!hasTrailingExpression && ((object)submissionResultType != null))
+                {
+                    Debug.Assert(submissionResultType.SpecialType != SpecialType.System_Void);
+
+                    var trailingExpression = new BoundDefaultOperator(method.GetNonNullSyntaxNode(), submissionResultType);
+                    var newStatements = block.Statements.Add(new BoundReturnStatement(trailingExpression.Syntax, trailingExpression));
+                    block = new BoundBlock(block.Syntax, ImmutableArray<LocalSymbol>.Empty, newStatements) { WasCompilerGenerated = true };
```

Comment:
Are we emitting an unreachable return if there are explicit returns in all code paths? Is that an issue?